src/components/tag-input/tag-input.component.ts

```diff
@@ -51,19 +51,10 @@ export class TagInputComponent<T = any> implements AfterContentInit, OnChanges,
     }
     set tags(value: ReadonlyArray<T>) {
         this._tags = value;
-        this._onChangeHandler(this._tags);
-        this.tagsChange.emit(this._tags);
     }
 
     /** The editable text appearing in the tag input. */
-    @Input()
-    get input() {
-        return this._input;
-    }
-    set input(value: string) {
-        this._input = value;
-        this.inputChange.emit(value);
-    }
+    @Input() input: string = '';

```

Comment:
Was there a reason not to use Subject here as well? To my understanding it's the same problem as with `value`.

Reply:
I can see it used in one place inside `keyHandler`:

````ts
const inputLength = this.input ? this.input.length : 0;
````

This is something that could be calculated in a subscription.

My issue is with code maintainability rather than whether this works or not.

Reply:
We also have the `commitInput` function which reads the `this.input` value are you happy to have this stored in a private variable as well?

Reply:
Ah, you're right, I missed that. Let's just leave it as it is then.